delete.js

```diff
@@ -1,33 +1,53 @@
-function buildContactList() {
-  const allAddress = document.getElementById('contact_details');
-  return Array.from(allAddress.rows);
-}
-
-function displayInfoMessage(userResult) {
-  const note = document.getElementById('error_message');
-  note.textContent = userResult;
-}
-
-function deleteContact(toBeDeleted) {
-  toBeDeleted.remove();
-}
-
-function contactDatabaseDeletion(nameInput) {
-  let rows = buildContactList();
-  let addressToDelete = rows.find(row => row.cells[0].textContent === nameInput.value);
-
-  if(addressToDelete !== undefined){
-    displayInfoMessage('');
-    deleteContact(addressToDelete);
-  }else {
-    displayInfoMessage('That contact does not exist');
+const addressBook = {
+  allAddress: document.getElementById('contact_details'),
+
+  buildContactList() {
+    return Array.from(this.allAddress.rows);
+  },
+
+  findContactToDelete(nameInput) {
+    let rows = this.buildContactList();
+    return rows.find(row => row.cells[0].textContent === nameInput);
+  },
+
+  deleteContact(toBeDeleted) {
+    toBeDeleted.remove();
   }
-}
+};
+
+const deletionUi = {
+  _userEntry: document.getElementById('user_contact_name'),
+  _messageOutput: document.getElementById('error_message'),
+
+  get userEntry() {
+    return this._userEntry.value;
+  },
+
+  set displayInfoMessage(userResult) {
```

Comment:
This method, together with `addressBook.allAddress` and `addressBook.buildContactList()` seems to be used all across the different UIs... what about putting them in a base UI so that all the other UIs can share it? 

For example, it doesn't make much sense to call the `deletionUI` (which is about deletions) from inside the `searcherUI` (which is about searches), just because you want to print an error message.

It would make sense to call the base UI though! :+1: